### ISSUE TITLE
Importing a network from an xarray fails due to a bad log string

### ISSUE THREAD

**Issue by osiccha** [PyPSA/PyPSA#381]
## Checklist

- [ ] I am using the current [`master`](https://github.com/pypsa/pypsa/tree/master) branch or the latest [release](https://github.com/pypsa/pypsa/releases). Please indicate: using release 0.19.2

## Describe the Bug

Importing a network using `import_from_netcdf` from an xarray instead of the path to the file raises a TypeError. The bug occurs in
`pypsa/io.py`. The function `import_from_netcdf` calls `_import_from_importer` with the `basename` argument being `None` instead of a string. When `_import_from_importer` tries to make the log message at the end, it tries to concatenate `basename` into the log string, which fails due to it being `None`. Loading itself works if making the log string is commented out or the basename is changed to an empy string.

A minimal example for this error:
```
import pypsa
testNetwork = pypsa.Network()
testNetwork.add("Bus", "bus")
exportedXarray = testNetwork.export_to_netcdf()

loadedNetwork = pypsa.Network()
loadedNetwork.import_from_netcdf(exportedXarray)
```

## Error Message

```
TypeError                                 Traceback (most recent call last)
/tmp/ipykernel_971/92175138.py in <module>
      4 
      5 loadedNetwork = pypsa.Network()
----> 6 loadedNetwork.import_from_netcdf(exportedXarray)

~/lib/python3.8/site-packages/pypsa/io.py in import_from_netcdf(network, path, skip_time)
    524     basename = os.path.basename(path) if isinstance(path, str) else None
    525     with ImporterNetCDF(path=path) as importer:
--> 526         _import_from_importer(network, importer, basename=basename,
    527                               skip_time=skip_time)
    528 

~/lib/python3.8/site-packages/pypsa/io.py in _import_from_importer(network, importer, basename, skip_time)
    665         imported_components.append(list_name)
    666 
--> 667     logger.info("Imported network{} has {}".format(" " + basename, ", ".join(imported_components)))
    668 
    669 def import_components_from_dataframe(network, dataframe, cls_name):

TypeError: can only concatenate str (not "NoneType") to str
```

**Comment by euronion** [comment]
We could either change
* `basename` to `network.name`, or
* `basename` to `str(basename or "")`

the second one would preseve the current code while printing an empty string in the described "bug" case.

**Comment by osiccha** [comment]
Because the log for importing from a file includes the name of the file, it would make sense for the import using an xarray to take the closest thing, which is either just saying that it was an xarray, or use the network name inside it.

**Comment by euronion** [comment]
So `str(basename or network.name)` then?

**Comment by osiccha** [comment]
I think either that to get the name of the network we load it into, or 
`basename = os.path.basename(path) if isinstance(path, str) else path.network_name` in the body of `import_from_netcdf` which
would return the name of the network saved in the xarray.

**Comment by euronion** [comment]
Since the Importer is for importing from NetCDF and importing from `xarray` is an undocumented feature, I prefer that we make the code handle empty network names in general on a lower level instead of handling `path` as `str or pathlib.Path` or `xarray.Dataset`.

I've prepared a fix over in #386 , have a look.
